src/histolab/filters/image_filters_functional.py

```diff
@@ -0,0 +1,744 @@
+    np.ndarray
+        NumPy boolean array representing the mask based on local Otsu threshold
+    """
+    if img.size != 2:
+        raise ValueError("Input must be 2D.")
+    img_arr = np.array(img)
+    local_otsu = sk_filters.rank.otsu(img_arr, sk_morphology.disk(disk_size))
+    return local_otsu
+
+
+def filter_entropy(
+    img: PIL.Image.Image, neighborhood: int = 9, threshold: int = 5
+) -> np.ndarray:
+    """Filter image based on entropy (complexity).
+
+    Note that input must be 2D.
+
+    Parameters:
+    ----------
+    img: PIL.Image.Image
+        input 2-dimensional image
+    neighborhood : int, optionale (default is 9)
+        Neighborhood size (defines height and width of 2D array of 1's).
+    threshold: Threshold value.
+
+    Returns
+    -------
+    np.ndarray
+        NumPy boolean array where True represent a measure of complexity.
+    """
+    if img.size != 2:
+        raise ValueError("Input must be 2D.")
+    img_arr = np.array(img)
+    entropy = sk_filters.rank.entropy(img_arr, np.ones((neighborhood, neighborhood)))
+    return _filter_threshold(entropy, threshold)
+
+
+# input of canny filter is a greyscale
+def canny_edges(
+    img: PIL.Image.Image,
+    sigma: float = 1.0,
+    low_threshold: float = 0.0,
+    high_threshold: float = 25.0,
+) -> np.ndarray:
+    """Filter image based on Canny algorithm edges.
+
+    Note that input image must be 2D
+
+    Parameters
+    ----------
+    img : PIL.Image.Image
+        Input 2-dimensional image
+    sigma : float, optional (default is 1.0)
+        Width (std dev) of Gaussian.
+    low_threshold : float, optional (default is 0.0)
+        Low hysteresis threshold value.
+    high_threshold : float, optional (default is 25.0)
+        High hysteresis threshold value.
+
+    Returns
+    -------
+    np.ndarray
+        Boolean NumPy array representing Canny edge map.
+    """
+    if img.size != 2:
+        raise ValueError("Input must be 2D.")
+    img_arr = np.array(img)
+    canny_edges = sk_feature.canny(
+        img_arr,
+        sigma=sigma,
+        low_threshold=low_threshold,
+        high_threshold=high_threshold,
+    )
+    return canny_edges
+
+
+def grays(img: PIL.Image.Image, tolerance: int = 15) -> np.ndarray:
+    """Filter out gray pixels, that is  pixels where the red, green,
+    and blue channel values are similar, i.e. under a specified tolerance.
+
+    Parameters
+    ----------
+    img : PIL.Image.Image
+          Input image
+    tolerance: int, optional (default is 15)
+               if difference between values is below this threshold,
+               values are considered similar and thus filtered out
+
+    Returns
+    -------
+    PIL.Image.Image
+         Boolean image where the grays values are masked out
+    """
+    if img.size != 3:
+        raise ValueError("Input must be 3D")
+    # TODO: class image mode exception: raise exception if not RGB
+    img_arr = np.array(img)
+    img_arr = img_arr.astype(np.int)
```

Comment:
i think that line 447 and 448 could be collapsed onto 1 avoiding the cast operation in a separate line of code